internal/controller/bucket/objectlockconfiguration.go

```diff
@@ -0,0 +1,195 @@
+package bucket
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	s3types "github.com/aws/aws-sdk-go-v2/service/s3/types"
+	"github.com/aws/smithy-go/document"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+
+	"github.com/linode/provider-ceph/apis/provider-ceph/v1alpha1"
+	apisv1alpha1 "github.com/linode/provider-ceph/apis/v1alpha1"
+	"github.com/linode/provider-ceph/internal/backendstore"
+	"github.com/linode/provider-ceph/internal/consts"
+	"github.com/linode/provider-ceph/internal/controller/s3clienthandler"
+	"github.com/linode/provider-ceph/internal/otel/traces"
+	"github.com/linode/provider-ceph/internal/rgw"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"go.opentelemetry.io/otel"
+)
+
+// ObjectLockConfigurationClient is the client for API methods and reconciling the ObjectLockConfiguration
+type ObjectLockConfigurationClient struct {
+	backendStore    *backendstore.BackendStore
+	s3ClientHandler *s3clienthandler.Handler
+	log             logging.Logger
+}
+
+func NewObjectLockConfigurationClient(b *backendstore.BackendStore, h *s3clienthandler.Handler, l logging.Logger) *ObjectLockConfigurationClient {
+	return &ObjectLockConfigurationClient{backendStore: b, s3ClientHandler: h, log: l}
+}
+
+func (l *ObjectLockConfigurationClient) Observe(ctx context.Context, bucket *v1alpha1.Bucket, backendNames []string) (ResourceStatus, error) {
+	ctx, span := otel.Tracer("").Start(ctx, "bucket.ObjectLockConfigurationClient.Observe")
+	defer span.End()
+
+	if bucket.Spec.ForProvider.ObjectLockEnabledForBucket == nil || !*bucket.Spec.ForProvider.ObjectLockEnabledForBucket {
+		l.log.Info("Object lock configuration not enabled in Bucket CR", consts.KeyBucketName, bucket.Name)
+
+		return Updated, nil
```

Comment:
It makes no difference when we are calling `Observe`, because we only care about `NeedsDeletion` or `NeedsUpdate`. 
The difference between `Updated` and `NoAction` is only relevant for some subresources when we are in the `Handle` phase.